### ISSUE TITLE
conditional row (not cell) formatting

### ISSUE THREAD

**Issue by liquidaty** [handsontable/handsontable#5380]
I made a relatively minor change (2 lines added, 2 lines changed) to accommodate row-level class formatting. This is similar to what is illustrated at https://docs.handsontable.com/5.0.1/demo-conditional-formatting.html, except that it provides a means to change the class at the TR level rather than the TD level, which is sometimes more desirable (and perhaps more performant than adding a class to every TD).

This modification is a bit of a hack in that it (ab)uses the existing cellRenderer function and just passes it the TR to do what it wants with it. The TR is only passed the first time cellRenderer is called for any given row, in order to avoid duplicative TR processing. I'd be happy to create a pull request reflecting these changes, if you'd like.

With this change, a custom renderer can modify the TR as follows:

    function my_renderer(_instance, _td, row, _col, _prop, _value, cellProps) {
      if(cellProps.tr && my_criteria_is_true(row))
        $(cellProps.tr).addClass('my_conditionally_added_class');
    }

As far as I can tell this approach works fine, but if you see any issues with this approach please let me know.

This is similar to the first answer of [this SO post](https://stackoverflow.com/questions/21443229/conditionally-highlight-row-in-handsontable-based-on-a-cell-value), but would only modify the TR once per row rather than multiple times per row.

[tr_formatting.diff.txt](https://github.com/handsontable/handsontable/files/2329932/tr_formatting.diff.txt)

**Comment by AMBudnik** [comment]
Hi @liquidaty 

as far as I understand you'd like to add a new parameter to the renderer function that will refer to the row (like a set of cells, or an array of cells to be more precise). That can actually help a lot of users. I have myself been asked many times to perform actions for the whole row/record.

ps. we're using vanilla js

**Comment by liquidaty** [comment]
Yes, that is correct, but the change addresses a more specific need than that.

It is easy to get the row already. One can just do `td.closest('tr')`. However, it is not efficient to use that approach, because then I am processing the TR for every TD, so if I have 20 visible columns, the code ends up executing 20 times more than it should. 

The change I made addresses this problem by only setting the TR in cellProperties once per row render. This allows much cleaner TR-handling code, and relative to the above alternative, significantly reduces the number of unnecessary iterations (I would imagine, by at least 90% on average if the average table has >= 10 columns).

In theory, this enhancement could be further optimized to set cellProperties.tr when tr actually changed (e.g. don't set cellProperties.tr if the user is scrolling horizontally), but I don't think that further optimization is necessary or even desirable given the marginal performance difference, the edge applicable use case and the related increase in code volume/complexity.

**Comment by AMBudnik** [comment]
Sure, we welcome all pull request. Just please make sure to
1. Sign CLA
2. Use `develop` branch
3. Add proper tests

**Comment by jmanos** [comment]
@liquidaty @AMBudnik 

This is a feature I would like to see.

However, sometimes the criteria for the row's class needs to be based on a different column than the first one in the table. Therefore, I would recommend including the tr property for all columns in the row. The user can then decide if only the first column should be used:

```
function my_renderer(_instance, _td, row, col, _prop, _value, cellProps) {
  if(cellProps.visualCol == 0 && other_criteria_is_true) {
    cellProps.tr.classList.add('my_conditionally_added_class');
}
```

The user could potentially have duplicative row processing if they do not handle the rows correctly, but it is more flexible and meets more use cases.

**Comment by AMBudnik** [comment]
Thank you for sharing your proposition @jmanos
